### PR TITLE
[feature](Azure) Implement generate_presigned_url function for Azure object client

### PR DESCRIPTION
**Pull request by ByteYue**
## Proposed changes

implement generate_presigned_url function for azure object client.

The implementation is inspired by the [sample](https://github.com/Azure/azure-sdk-for-cpp/blob/main/sdk/storage/azure-storage-blobs/samples/blob_sas.cpp) code of azure sdk

### CODE REVIEW

File: be/src/io/fs/azure_obj_storage_client.cpp [hunk 1]
```diff
@@ -251,4 +256,21 @@ ObjectStorageResponse AzureObjStorageClient::delete_objects_recursively(
...
+std::string AzureObjStorageClient::generate_presigned_url(const ObjectStoragePathOptions& opts,
+                                                          int64_t expiration_secs,
+                                                          const S3ClientConf& conf) {
+    Azure::Storage::Sas::BlobSasBuilder sas_builder;
+    sas_builder.ExpiresOn =
+            std::chrono::system_clock::now() + std::chrono::seconds(expiration_secs);
+    sas_builder.BlobContainerName = opts.bucket;
+    sas_builder.BlobName = opts.key;
+    sas_builder.Resource = Azure::Storage::Sas::BlobSasResource::Blob;
+    sas_builder.SetPermissions(Azure::Storage::Sas::BlobSasPermissions::Write);
```
Comment (gavinchou) [inline be/src/io/fs/azure_obj_storage_client.cpp:269]: We also need "Read" permission here.
Make sure that this signature is constrained by the specific blob.

And USE HTTPS only here
```
sas_biulder.Protocol = SasProtocol.HttpsOnly;
```

**Review by gavinchou — CHANGES_REQUESTED** [review]

**Review by liaoxin01 — DISMISSED** [review]
LGTM

File: be/src/io/fs/azure_obj_storage_client.cpp [hunk 2]
```diff
@@ -251,4 +257,23 @@ ObjectStorageResponse AzureObjStorageClient::delete_objects_recursively(
...
+                                                          const S3ClientConf& conf) {
+    Azure::Storage::Sas::BlobSasBuilder sas_builder;
+    sas_builder.ExpiresOn =
+            std::chrono::system_clock::now() + std::chrono::seconds(expiration_secs);
+    sas_builder.BlobContainerName = opts.bucket;
+    sas_builder.BlobName = opts.key;
+    sas_builder.Resource = Azure::Storage::Sas::BlobSasResource::Blob;
+    sas_builder.Protocol = Azure::Storage::Sas::SasProtocol::HttpsOnly;
+    sas_builder.SetPermissions(Azure::Storage::Sas::BlobSasPermissions::Read |
+                               Azure::Storage::Sas::BlobSasPermissions::Write);
```
Comment (gavinchou) [inline be/src/io/fs/azure_obj_storage_client.cpp]: No need `write` permission

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

File: cloud/src/recycler/obj_store_accessor.h [hunk 3]
```diff
@@ -97,6 +97,7 @@ struct ObjectStorageResponse {
     int ret {0}; // To unify the error handle logic with BE, we'd better use the same error code as BE
     // clang-format on
     std::string error_msg;
+    static ObjectStorageResponse OK() { return {}; }
```
Comment (gavinchou) [inline cloud/src/recycler/obj_store_accessor.h:100]: `return {};` -> return explicit OK  not the default one.

**Review by gavinchou — CHANGES_REQUESTED** [review]

**Comment by ByteYue** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Review by platoneko — APPROVED** [review]

**Pull request merged**